### PR TITLE
[SYCL] string.h function test for libdevice

### PR DESCRIPTION
**Pull request by jinge90**
Signed-off-by: gejin <ge.jin@intel.com>

This PR adds test for functions in <string.h> including memcpy, memset, memcmp. Once https://github.com/intel/llvm/pull/4147
is merged, this test should pass.
Thanks very much.

### CODE REVIEW

File: SYCL/DeviceLib/string_test.cpp [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+                     sycl::access::placeholder::false_t>
+          dst_acc(buffer2, cgh);
+
+      sycl::accessor<char, 1, sycl::access::mode::write,
+                     sycl::access::target::global_buffer,
+                     sycl::access::placeholder::false_t>
+          dst1_acc(buffer3, cgh);
+      cgh.single_task<class KernelTestMemcpyAddrSpace>([=]() {
+        // memcpy from constant buffer to local buffer
+        memcpy(local_acc.get_pointer(), src_acc.get_pointer(), 8);
```
Comment (jinge90) [inline SYCL/DeviceLib/string_test.cpp:237]: Hi, @vzakhari 
kernel_test_memcpy_addr_space is used to test memcpy whose input pointers have different address space. Could you help review if it meets our requirement?
Thanks very much.

**Review by vzakhari — DISMISSED** [review]

**Comment by vladimirlaz** [comment]
@jinge90 could you please fix clang-format issues?

**Review by vladimirlaz — CHANGES_REQUESTED** [review]
@jinge90, could you please fix formatting issues and make sure the test is passing on all configurations?

**Comment by jinge90** [comment]
Hi, vladimirlaz
I have fixed the clang-format issue but found a lot of CUDA tests have failed. Adding a test case shouldn't impact other tests, so is there any other problem?
Thanks very much.

**Comment by vladimirlaz** [comment]
> Hi, vladimirlaz
> I have fixed the clang-format issue but found a lot of CUDA tests have failed. Adding a test case shouldn't impact other tests, so is there any other problem?
> Thanks very much.
This is known issue addressed in intel/llvm#4348.

**Review by vladimirlaz — APPROVED** [review]

**Review by vzakhari — APPROVED** [review]

**Pull request merged**